src/sparseml/openpifpaf/train.py

```diff
@@ -151,11 +152,35 @@ def main():
 
     datamodule = openpifpaf.datasets.factory(args.dataset)
 
+    """
+    Cases for checkpointing:
+    1. No checkpoint - no changes to base flow
+    2. Checkpoint from local file - no changes to base flow
+    3. Checkpoint from zoo stub:
+        a. set args.checkpoint to None to get a base net
+        b. manually load in state dict
+    """
+    zoo_stub = None
+    if args.checkpoint and args.checkpoint.startswith("zoo:"):
+        checkpoint_path = download_framework_model_by_recipe_type(Model(zoo_stub))
+        checkpoint = torch.load(checkpoint_path, map_location="cpu")
+        args.basenet = checkpoint["meta"]["args"]["basenet"]
+        LOG.info(f"Overriding --basenet with value from checkpoint: {args.basenet}")
+        zoo_stub = args.checkpoint
+        args.checkpoint = None
+
+    # NOTE: configure this after removing zoo stub checkpoint
+    openpifpaf.network.Factory.configure(args)
+
     net_cpu, start_epoch = openpifpaf.network.Factory().factory(
         head_metas=datamodule.head_metas
     )
     loss = openpifpaf.network.losses.Factory().factory(datamodule.head_metas)
 
+    if zoo_stub is not None:
+        # just load state dict from zoo stub
+        net_cpu.load_state_dict(checkpoint["state_dict"])
```

Comment:
calling into `load_state_dict` immediately after model initialization isn't safe since if the model was structurally changed in a previous run, the module state will not state dict keys. we should connect this with the logic in `_load_managers_from_checkpoint` so the flow includes this